server/src/main/java/org/elasticsearch/search/aggregations/bucket/terms/LongKeyedBucketOrds.java

```diff
@@ -101,6 +106,45 @@ private LongKeyedBucketOrds() {}
      */
     public abstract BucketOrdsEnum ordsEnum(long owningBucketOrd);
 
+    public Iterator<Long> keyOrderedIterator(long owningBucketOrd) {
+        if (keySet == null) {
+            keySet = new TreeSet<>();
```

Comment:
We could also build this as we're adding values, but since add is very hot, it seemed safer to do it out of that loop.

Reply:
We also don't always call `keyOrderedIterator`.